src/lib/api.ts

```diff
@@ -25,6 +25,15 @@ export const getExistingUser = async (
     },
   )
 
+  /**
+   * fetch needs to be rejected manually to trigger react-query error
+   * @see https://tanstack.com/query/v4/docs/react/guides/query-functions#usage-with-fetch-and-other-clients-that-do-not-throw-by-default
+   */
+  if (!rawRes.ok) {
+    const error = await rawRes.json()
+    throw new Error(error.message)
+  }
+
```

Comment:
you should consider using axios instead for easier handling 🥲 @mazipan

Reply:
Nope, I will still using fetch.
I will try to tidy up the code later, so it will handle this kind of Error throwing as well.

Thanks for pointing out 🙌